snapcraft/plugins/_python/_python_finder.py

```diff
@@ -0,0 +1,44 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import glob
+
+
+def get_python_command(python_version, stage_dir, install_dir):
+    python_command = os.path.join('usr', 'bin', python_version)
+
+    # staged as in the stage dir, not from stage-packages
+    staged_python = os.path.join(stage_dir, python_command)
+    unstaged_python = os.path.join(install_dir, python_command)
+
+    if os.path.exists(staged_python):
+        return staged_python
+    else:
+        return unstaged_python
+
+
+def get_python_headers(python_version, stage_dir):
+    base_match = os.path.join('usr', 'include', '{}*'.format(python_version))
+    unstaged_python = glob.glob(os.path.join(os.path.sep, base_match))
+    staged_python = glob.glob(os.path.join(stage_dir, base_match))
+
+    if staged_python:
+        return staged_python[0]
+    elif unstaged_python:
+        return unstaged_python[0]
+    else:
+        return ''
```

Comment:
Wouldn't it be more pythonic to raise an exception?

Reply:
I would agree if it was actually an error, but it's not unexpected in the code. If I raised an exception here, the usage of this function would look like:

```
try:
    headers = get_python_headers([...])
except MissingPythonHeadersError:
    pass
else:
    # Do something with the headers
```

Right now it's just an if, which feels a little cleaner.